projects/core/src/routing/event/routing.events.ts

```diff
@@ -0,0 +1,70 @@
+import { Images } from '../../model';
+import {
+  BaseOption,
+  Category,
+  Classification,
+  FutureStock,
+  Price,
+  PriceRange,
+  Product,
+  ProductReferences,
+  Promotion,
+  Review,
+  Stock,
+  VariantMatrixElement,
+  VariantOption,
+  VariantType,
+} from '../../model/product.model';
+
+export class PageVisited {}
```

Comment:
Maybe we can remove this "abstract" class, as it doesn't contain any common properties.
Or do you think we can add something here?

Reply:
I removed the `extends PageVisited ` part for the other classes. There is also a generic "PageVisited" event